### PR TITLE
PLATOPS-1091 - How to upgrade to the latest version

### PR DESCRIPTION
**Pull request by jakobgrunig**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,6 +9,49 @@ This started as a fork of [Play-ReactiveMongo](https://github.com/ReactiveMongo/
...
+```No Json serializer as JsObject found for type reactivemongo.bson.BSONDocument. Try to implement an implicit OWrites or OFormat for this type.```
+Try adding the following import
+```scala
+import reactivemongo.play.json.ImplicitBSONHandlers._
+```
+
+#### WriteResult is no longer an Exception
+
+The type hierarchy of the trait `WriteResult` has changed in new version of reactivemongo. 
+It’s no longer an Exception. As it now longer represents errors in the public API, the following properties have been removed: `errmsg`, `hasErrors`, `inError` and `message`.
```
Comment (githubmo) [inline README.md]: `As it no longer` instead of `As it now longer`

Comment (githubmo) [reply README.md]: `errmsg`, `hasErrors` and `inError` are now kept in protective scope and are internal state kept encapsulated. If you wish to know whether the command is in error or not, you would have to check for `ok`.

File: README.md [hunk 2]
```diff
@@ -9,6 +9,49 @@ This started as a fork of [Play-ReactiveMongo](https://github.com/ReactiveMongo/
...
+    case _ => // return successfully
+}
+```
+
+Now, you have to do this:
+```scala
+collection.doSomething([...]) map {
+    case _ => // return successfully
+} recover {
+   case lastError => // handle the error
```
Comment (githubmo) [inline README.md]: Might be worth changing this to:

```
collection.doSomething([...]) map {
     case _ => // return successfully
 } recover {
    case lastError: LastError => // Handle DB specific issue
    case e => // Handle other exceptions
}
```

File: README.md [hunk 3]
```diff
@@ -9,6 +9,49 @@ This started as a fork of [Play-ReactiveMongo](https://github.com/ReactiveMongo/
...
+```No Json serializer as JsObject found for type reactivemongo.bson.BSONDocument. Try to implement an implicit OWrites or OFormat for this type.```
+Try adding the following import
+```scala
+import reactivemongo.play.json.ImplicitBSONHandlers._
+```
+
+#### WriteResult is no longer an Exception
+
+The type hierarchy of the trait `WriteResult` has changed in new version of reactivemongo. 
+It’s no longer an Exception. As it no longer represents errors in the public API, the following properties have been removed: `errmsg`, `hasErrors`, `inError` and `message`.
```
Comment (githubmo) [inline README.md]: Don't know if you want to change the wordings around this as they're not removed, just encapsulated. It is `protected` in `WriteResult` but some of it's implementation elevate it to public.

File: README.md (hunk 2 above)
Comment (jakobgrunig) [reply README.md]: The example of the old version wasn't. I prefer to keep focus on the use of lastError

File: README.md (hunk 3 above)
Comment (jakobgrunig) [reply README.md]: I copied the sentence from the official documentation. If it is not accessible on the public API, it is like removed

File: README.md (hunk 2 above)
Comment (githubmo) [reply README.md]: Ok, either way the main point is given across.

File: README.md (hunk 3 above)
Comment (githubmo) [reply README.md]: Ok, fair enough.

**Comment by githubmo** [comment]
LGTM 👍

**Pull request merged**